internal/app/secretless/handlers/mysql/protocol/protocol.go

```diff
@@ -711,9 +711,11 @@ func CheckPacketLength(expected int, packet []byte) error {
 // https://dev.mysql.com/doc/internals/en/secure-password-authentication.html#packet-Authentication::Native41
 // SHA1( password ) XOR SHA1( "20-bytes random data from server" <concat> SHA1( SHA1( password ) ) )
 func NativePassword(password string, salt []byte) (nativePassword []byte, err error) {
-
+	return NativePasswordWithBytes([]byte(password), salt)
```

Comment:
Rather than doing this, can we just update where it's called to pass in a byte array? It's just [one place](https://github.com/cyberark/secretless-broker/blob/master/internal/app/secretless/handlers/mysql/protocol/protocol.go#L513) in the code and [one test](https://github.com/cyberark/secretless-broker/blob/master/internal/app/secretless/handlers/mysql/protocol/protocol_test.go#L429) that would need to be updated. I don't see the value in keeping the original method with the string input if we are trying to get away from that - in fact, I see the very clear drawback that if we leave it here, someone may use it as we're trying to get away from passing strings around.